client-java/controller/src/main/java/org/evomaster/client/java/controller/opensearch/OpenSearchHeuristicsCalculator.java

```diff
@@ -103,6 +294,463 @@ private double compareValues(Object val1, Object val2) {
+        if (!((Map<?,?>) doc).containsKey(field)) {
+            return Double.MAX_VALUE;
+        }
+        
+        Object actualValue = ((Map<?,?>) doc).get(field);
+        String actualStr = actualValue.toString();
+        
+        // Calculate edit distance (Levenshtein distance)
+        int editDistance = calculateEditDistance(actualStr, expectedValue, transpositions != null ? transpositions : true);
+        
+        // If fuzziness is specified, check if within allowed distance
+        if (fuzziness != null) {
+            if (editDistance <= fuzziness) {
+                return 0.0;
+            } else {
+                return editDistance - fuzziness;
+            }
+        }
+        
+        // Default fuzziness is AUTO, which is typically 0, 1, or 2 based on term length
+        int defaultFuzziness = getAutoFuzziness(expectedValue.length());
+        if (editDistance <= defaultFuzziness) {
+            return 0.0;
+        }
+        
+        return editDistance - defaultFuzziness;
+    }
+
+    /**
+     * Calculate distance for Wildcard operation.
+     * Wildcard query matches if the document field value matches the wildcard pattern.
+     */
+    private double calculateDistanceForWildcard(WildcardOperation operation, Object doc) {
+        String field = operation.getFieldName();
+        String pattern = operation.getValue();
+        Boolean caseInsensitive = operation.getCaseInsensitive();
+        
+        if (!((Map<?,?>) doc).containsKey(field)) {
+            return Double.MAX_VALUE;
+        }
+        
+        Object actualValue = ((Map<?,?>) doc).get(field);
+        String actualStr = actualValue.toString();
+        
+        // Apply case insensitive comparison if needed
+        if (caseInsensitive != null && caseInsensitive) {
+            actualStr = actualStr.toLowerCase();
+            pattern = pattern.toLowerCase();
+        }
+        
+        if (matchesWildcard(actualStr, pattern)) {
+            return 0.0;
+        }
+        
+        // Calculate approximate distance based on pattern similarity
+        return (double) DistanceHelper.getLeftAlignmentDistance(actualStr, pattern.replace("*", "").replace("?", ""));
+    }
+
+    /**
+     * Calculate distance for Regexp operation.
+     * Regexp query matches if the document field value matches the regular expression.
+     */
+    private double calculateDistanceForRegexp(RegexpOperation operation, Object doc) {
+        String field = operation.getFieldName();
+        String regex = operation.getValue();
+        Boolean caseInsensitive = operation.getCaseInsensitive();
+        
+        if (!((Map<?,?>) doc).containsKey(field)) {
+            return Double.MAX_VALUE;
+        }
+        
+        Object actualValue = ((Map<?,?>) doc).get(field);
+        String actualStr = actualValue.toString();
+        
+        try {
+            // Create pattern with case insensitive flag if needed
+            java.util.regex.Pattern pattern;
```

Comment:
are the regex in OpenSearch having the same syntax of Java regex? clarify here in some comments, with possible some links